### PR TITLE
Feature/add to cart and fetch cart

### PR DESCRIPTION
**Pull request by rakeshkumar-verma**

### CODE REVIEW

File: core/src/main/java/com/hcl/ecomm/core/services/AddToCartService.java [hunk 1]
```diff
@@ -0,0 +1,18 @@
+package com.hcl.ecomm.core.services;
```
Comment (DeepaliAgrwl) [inline core/src/main/java/com/hcl/ecomm/core/services/AddToCartService.java:1]: @rakeshkumar-verma why would we need separate class for this ? can't we merge it into cartService ?

File: core/src/main/java/com/hcl/ecomm/core/services/CreateCartService.java [hunk 2]
```diff
@@ -0,0 +1,13 @@
+package com.hcl.ecomm.core.services;
```
Comment (DeepaliAgrwl) [inline core/src/main/java/com/hcl/ecomm/core/services/CreateCartService.java:1]: @rakeshkumar-verma  can't we have a single service for all cart related functions ?

File: core/src/main/java/com/hcl/ecomm/core/services/DeleteCartItemService.java [hunk 3]
```diff
@@ -0,0 +1,13 @@
+package com.hcl.ecomm.core.services;
```
Comment (DeepaliAgrwl) [inline core/src/main/java/com/hcl/ecomm/core/services/DeleteCartItemService.java]: @rakeshkumar-verma  can't we have a single service for all cart related functions ?

Comment (rakeshkumar-verma) [reply core/src/main/java/com/hcl/ecomm/core/services/DeleteCartItemService.java]: @rishabh-behl  @DeepaliAgrwl  Ideally we should have only one service for Magento but all working parallely to hard to track. We can do this in code cleanup.

File: core/src/main/java/com/hcl/ecomm/core/services/CreateCartService.java (hunk 2 above)
Comment (rakeshkumar-verma) [reply core/src/main/java/com/hcl/ecomm/core/services/CreateCartService.java:1]: @rishabh-behl @DeepaliAgrwl Ideally we should have only one service for Magento but all working parallely to hard to track. We can do this in code cleanup.

File: core/src/main/java/com/hcl/ecomm/core/services/impl/CreateCartServiceImpl.java [hunk 4]
```diff
@@ -0,0 +1,106 @@
...
+
+import com.hcl.ecomm.core.config.MagentoServiceConfig;
+import com.hcl.ecomm.core.services.CreateCartService;
+import com.hcl.ecomm.core.services.LoginService;
+
+@Component(
+		immediate = true,
+		enabled = true, 
+		service = CreateCartService.class)
+@Designate(ocd = MagentoServiceConfig.class)
```
Comment (DeepaliAgrwl) [inline core/src/main/java/com/hcl/ecomm/core/services/impl/CreateCartServiceImpl.java]: @rakeshkumar-verma  why we have AddToCartServiceConfig in code base when we are using MagentoServiceConfig class for the configurations

File: core/src/main/java/com/hcl/ecomm/core/services/impl/DeleteCartItemServiceImpl.java [hunk 5]
```diff
@@ -0,0 +1,105 @@
...
+
+import com.hcl.ecomm.core.config.MagentoServiceConfig;
+import com.hcl.ecomm.core.services.DeleteCartItemService;
+import com.hcl.ecomm.core.services.LoginService;
+
+@Component(
+		immediate = true,
+		enabled = true, 
+		service = DeleteCartItemService.class)
+@Designate(ocd = MagentoServiceConfig.class)
```
Comment (DeepaliAgrwl) [inline core/src/main/java/com/hcl/ecomm/core/services/impl/DeleteCartItemServiceImpl.java]: @rakeshkumar-verma  we don't require to declare MagentoServiceConfig for everySeviceImpl s it is making duplicate configurations on felix console

File: ui.apps/src/main/content/jcr_root/apps/hclecomm/config.author.dev/com.hcl.ecomm.core.services.impl.LoginServiceImpl.xml [hunk 6]
```diff
@@ -0,0 +1,7 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
Comment (rishabh-behl) [inline ui.apps/src/main/content/jcr_root/apps/hclecomm/config.author.dev/com.hcl.ecomm.core.services.impl.LoginServiceImpl.xml]: @DeepaliAgrwl : why are we check in run modes as part of this PR ?

File: ui.apps/src/main/content/jcr_root/apps/hclecomm/config.author.dev/com.hcl.ecomm.core.services.impl.ProductServiceImpl.xml [hunk 7]
```diff
@@ -0,0 +1,7 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
Comment (rishabh-behl) [inline ui.apps/src/main/content/jcr_root/apps/hclecomm/config.author.dev/com.hcl.ecomm.core.services.impl.ProductServiceImpl.xml]: @DeepaliAgrwl : why are we check in run modes as part of this PR ?

File: ui.apps/src/main/content/jcr_root/apps/hclecomm/config.author.qa/com.hcl.ecomm.core.services.impl.LoginServiceImpl.xml [hunk 8]
```diff
@@ -0,0 +1,7 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
Comment (rishabh-behl) [inline ui.apps/src/main/content/jcr_root/apps/hclecomm/config.author.qa/com.hcl.ecomm.core.services.impl.LoginServiceImpl.xml]: @DeepaliAgrwl : why are we check in run modes as part of this PR ?

File: ui.apps/src/main/content/jcr_root/apps/hclecomm/config.author.qa/com.hcl.ecomm.core.services.impl.ProductServiceImpl.xml [hunk 9]
```diff
@@ -0,0 +1,7 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
Comment (rishabh-behl) [inline ui.apps/src/main/content/jcr_root/apps/hclecomm/config.author.qa/com.hcl.ecomm.core.services.impl.ProductServiceImpl.xml]: @DeepaliAgrwl : why are we check in run modes as part of this PR ?

File: ui.apps/src/main/content/jcr_root/apps/hclecomm/config.author.dev/com.hcl.ecomm.core.services.impl.LoginServiceImpl.xml (hunk 6 above)
Comment (DeepaliAgrwl) [reply ui.apps/src/main/content/jcr_root/apps/hclecomm/config.author.dev/com.hcl.ecomm.core.services.impl.LoginServiceImpl.xml]: removed ..

File: ui.apps/src/main/content/jcr_root/apps/hclecomm/config.author.dev/com.hcl.ecomm.core.services.impl.ProductServiceImpl.xml (hunk 7 above)
Comment (DeepaliAgrwl) [reply ui.apps/src/main/content/jcr_root/apps/hclecomm/config.author.dev/com.hcl.ecomm.core.services.impl.ProductServiceImpl.xml]: removed.

File: ui.apps/src/main/content/jcr_root/apps/hclecomm/config.author.qa/com.hcl.ecomm.core.services.impl.LoginServiceImpl.xml (hunk 8 above)
Comment (DeepaliAgrwl) [reply ui.apps/src/main/content/jcr_root/apps/hclecomm/config.author.qa/com.hcl.ecomm.core.services.impl.LoginServiceImpl.xml]: removed.

File: ui.apps/src/main/content/jcr_root/apps/hclecomm/config.author.qa/com.hcl.ecomm.core.services.impl.ProductServiceImpl.xml (hunk 9 above)
Comment (DeepaliAgrwl) [reply ui.apps/src/main/content/jcr_root/apps/hclecomm/config.author.qa/com.hcl.ecomm.core.services.impl.ProductServiceImpl.xml]: removed.

File: ui.apps/src/main/content/jcr_root/apps/hclecomm/components/content/productdetails/clientlibs/js/detailsofproduct.js [hunk 10]
```diff
@@ -1,3 +1,4 @@
+var sku = "24-WG087";
```
Comment (rishabh-behl) [inline ui.apps/src/main/content/jcr_root/apps/hclecomm/components/content/productdetails/clientlibs/js/detailsofproduct.js]: @ashishgoyalweb : why is this hard-coded. Even if we this is an assignation it should be ""

Comment (DeepaliAgrwl) [reply ui.apps/src/main/content/jcr_root/apps/hclecomm/components/content/productdetails/clientlibs/js/detailsofproduct.js]: As per the discussion with Rajamouli, The sku parameter will be coming from PLP page or home page when the user click on a product from that page.  Right now we don't have those pages that is why he hardcoded the sku value.

File: core/src/main/java/com/hcl/ecomm/core/config/MagentoServiceConfig.java [hunk 11]
```diff
@@ -26,12 +28,19 @@
...
+	String magentoService_addToCartPath() default addToCartPath;
+	
+	@AttributeDefinition(name = "UPDATE_ITEM_QUANTITY_SERVICE_PATH", description = "This is update cart item API path", defaultValue = updateCartItemPath, type = AttributeType.STRING)
+	String magentoService_updateCartItemPath() default updateCartItemPath;
+
+	@AttributeDefinition(name = "SERVICE_PATH", description = "This is create empty cart API path", defaultValue = emptyCartPah, type = AttributeType.STRING)
+	String magentoService_emptyCartPath() default emptyCartPah;
 	
-	@AttributeDefinition(name = "GUEST CART ITEM DELETE SERVICE PATH", description = "This is guest cart item delete API path", defaultValue = guestCartItemDeletePah, type = AttributeType.STRING)
-    String deleteCartService_guestCartItemDeletePath() default guestCartItemDeletePah;
```
Comment (rishabh-behl) [inline core/src/main/java/com/hcl/ecomm/core/config/MagentoServiceConfig.java:31]: @DeepaliAgrwl : this is used in /hcl/ecomm/core/services/impl/DeleteCartItemServiceImpl.java

File: core/src/main/java/com/hcl/ecomm/core/config/MagentoServiceConfig.java [hunk 12]
```diff
@@ -26,12 +28,19 @@
...
+	@AttributeDefinition(name = "GUEST_CART_ITEM_DELETE_SERVICE_PATH", description = "This is guest cart item delete API path", defaultValue = guestCartItemDeletePah, type = AttributeType.STRING)
+	String magentoService_guestCartItemDeletePath() default guestCartItemDeletePah;
 
-    @AttributeDefinition(name = "Update Cart Items Service Path", description = "This is the API path for getting cart item details", defaultValue = servicePathCartUpdateItems, type = AttributeType.STRING )
-    String cartUpdate_servicePath_string() default servicePathCartUpdateItems;
```
Comment (rishabh-behl) [inline core/src/main/java/com/hcl/ecomm/core/config/MagentoServiceConfig.java:34]: @DeepaliAgrwl : this is used in /hcl/ecomm/core/services/impl/CartServiceImpl.java

File: core/src/main/java/com/hcl/ecomm/core/config/MagentoServiceConfig.java (hunk 11 above)
Comment (DeepaliAgrwl) [reply core/src/main/java/com/hcl/ecomm/core/config/MagentoServiceConfig.java:31]: synced with deveploment now

File: core/src/main/java/com/hcl/ecomm/core/config/MagentoServiceConfig.java (hunk 12 above)
Comment (DeepaliAgrwl) [reply core/src/main/java/com/hcl/ecomm/core/config/MagentoServiceConfig.java:34]: synced with development now

**Review by rishabh-behl — APPROVED** [review]
LGTM

**Pull request merged**